### ISSUE TITLE
tst_put_vars.c and tst_elatefill.c are netCDF-4 only tests, and should be moved to nc_test4 from nc_test

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#738]
These two tests are straight-up netCDF-4 tests. They were contributed by users who did not follow the convention of putting netCDF-4 only tests in nc_test4.

I will move them.

**Comment by wkliao** [comment]
I think both test programs can be used to test classic file formats as well.
All you have to do is to add a loop of all formats on top.

**Comment by edhartnett** [comment]
Yes, unfortunately the elatefill one will not work with classic at this time.

**Comment by wkliao** [comment]
The fix for classic files is in #731

**Comment by edhartnett** [comment]
OK, that's awesome! Thanks!

**Comment by edhartnett** [comment]
Once I separated libsrc4 from libhdf5, these tests become an issue in this directory. When it is possible to build without hdf5, but with netCDF4 (and HDF4, or a user-defined format), these files become a configuration issue.

In HPC NetCDF I have moved them to the nc_test4 directory, and all the configuration problems go away. They can still be run on all formats, but since they are sometimes run on HDF5, they work best in nc_test4. nc_test is a good place for tests that never use HDF5.

I will put up a PR moving these tests. Also tst_atts.c and tst_diskless2.c need to be moved. tst_atts.c becomes nc_test4/tst_atts3.c.
